examples/server.rs

```diff
@@ -80,34 +82,41 @@ async fn echo(req: Request<Body>) -> Result<Response<Body>, hyper::Error> {
 }
 
 // Load public certificate from file.
-fn load_certs(filename: &str) -> io::Result<Vec<rustls::Certificate>> {
+fn load_certs(filename: &str) -> io::Result<Vec<CertificateDer>> {
     // Open certificate file.
     let certfile = fs::File::open(filename)
         .map_err(|e| error(format!("failed to open {}: {}", filename, e)))?;
     let mut reader = io::BufReader::new(certfile);
 
     // Load and return certificate.
-    let certs = rustls_pemfile::certs(&mut reader)
-        .map_err(|_| error("failed to load certificate".into()))?;
+    let certs = rustls_pemfile::certs(&mut reader).flat_map(|x| x);
     Ok(certs
         .into_iter()
-        .map(rustls::Certificate)
+        .map(Into::into)
         .collect())
 }
 
 // Load private key from file.
-fn load_private_key(filename: &str) -> io::Result<rustls::PrivateKey> {
+fn load_private_key(filename: &str) -> io::Result<PrivateKeyDer> {
     // Open keyfile.
     let keyfile = fs::File::open(filename)
         .map_err(|e| error(format!("failed to open {}: {}", filename, e)))?;
     let mut reader = io::BufReader::new(keyfile);
 
     // Load and return a single private key.
-    let keys = rustls_pemfile::rsa_private_keys(&mut reader)
-        .map_err(|_| error("failed to load private key".into()))?;
+    let keys: Vec<pki_types::PrivatePkcs1KeyDer<'static>> =
+        rustls_pemfile::rsa_private_keys(&mut reader)
+            .flat_map(|x| x)
+            .collect();
     if keys.len() != 1 {
         return Err(error("expected a single private key".into()));
     }
 
-    Ok(rustls::PrivateKey(keys[0].clone()))
+    // TODO: should PKCS#8 be supported?
+    Ok(PrivateKeyDer::Pkcs1(
```

Comment:
If you submit a version bump to rustls-pemfile, we can use the `private_key()` function here which should take care of everything.

Reply:
`private_key()` read not just PKCS#1 keys. Maybe that doesn't work under that context which is RSA based...?

Reply:
I don't think there's a particular reason to prefer RSA keys here.

Reply:
Done